cmd/skopeo/flag_test.go

```diff
@@ -152,30 +146,28 @@ func TestOptionalStringSet(t *testing.T) {
 	// is not called in any possible situation).
 	var globalOS, commandOS optionalString
 	actionRun := false
-	app := cli.NewApp()
-	app.EnableBashCompletion = true
-	app.Flags = []cli.Flag{
-		cli.GenericFlag{
-			Name:  "global-OS",
-			Value: newOptionalStringValue(&globalOS),
+
+	app := &cobra.Command{
+		Use: "app",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return nil
 		},
 	}
-	app.Commands = []cli.Command{{
-		Name: "cmd",
-		Flags: []cli.Flag{
-			cli.GenericFlag{
-				Name:  "command-OS",
-				Value: newOptionalStringValue(&commandOS),
-			},
-		},
-		Action: func(*cli.Context) error {
+
```

Comment:
(Non-blocking: If there should be an empty line here (I don’t think there has to be one), it should be one line below, separating the “global” `app` and `globalOS`, and the sub-command `cmd` and `commandOS`.)